extensions/elytron-security-oauth2/runtime/src/main/java/io/quarkus/elytron/security/oauth2/runtime/auth/QuarkusOAuth2IntrospectValidator.java

```diff
@@ -0,0 +1,291 @@
+        return new QuarkusOAuth2IntrospectValidator.Builder();
+    }
+
+    private final URL tokenIntrospectionUrl;
+    private final String clientId;
+    private final String clientSecret;
+    private final SSLContext sslContext;
+    private final HostnameVerifier hostnameVerifier;
+
+    QuarkusOAuth2IntrospectValidator(QuarkusOAuth2IntrospectValidator.Builder configuration) {
+        this.tokenIntrospectionUrl = Assert.checkNotNullParam("tokenIntrospectionUrl", configuration.tokenIntrospectionUrl);
+        this.clientId = Assert.checkNotNullParam("clientId", configuration.clientId);
+        this.clientSecret = Assert.checkNotNullParam("clientSecret", configuration.clientSecret);
+
+        // SSLContext is lazily provided so we remove the check
+        //        if (tokenIntrospectionUrl.getProtocol().equalsIgnoreCase("https")) {
+        //            Assert.checkNotNullParam("sslContext", configuration.sslContext);
+        //        }
+
+        this.sslContext = configuration.sslContext;
+        this.hostnameVerifier = configuration.hostnameVerifier;
+    }
+
+    @Override
+    public Attributes validate(BearerTokenEvidence evidence) throws RealmUnavailableException {
+        Assert.checkNotNullParam("evidence", evidence);
+
+        try {
+            // Use the provided SSL context or the default one: this allow Quarkus to didn't specify an SSL Context at build time.
+            SSLContext context = this.sslContext == null ? SSLContext.getDefault() : this.sslContext;
+            JsonObject claims = introspectAccessToken(this.tokenIntrospectionUrl,
+                    this.clientId, this.clientSecret, evidence.getToken(), context, this.hostnameVerifier);
+
+            if (isValidToken(claims)) {
+                return toAttributes(claims);
+            }
+        } catch (Exception e) {
+            throw log.tokenRealmOAuth2TokenIntrospectionFailed(e);
+        }
+
+        return null;
+    }
+
+    private boolean isValidToken(JsonObject claims) {
+        return claims != null && claims.getBoolean("active", false);
+    }
+
+    /**
+     * Introspects an OAuth2 Access Token using a RFC-7662 compatible endpoint.
+     *
+     * @param tokenIntrospectionUrl an {@link URL} pointing to a RFC-7662 compatible endpoint
+     * @param clientId the identifier of a client within the OAUth2 Authorization Server
+     * @param clientSecret the secret of the client
+     * @param token the access token to introspect
+     * @param sslContext the ssl context
+     * @param hostnameVerifier the hostname verifier
+     * @return a @{JsonObject} representing the response from the introspection endpoint or null if
+     */
+    private JsonObject introspectAccessToken(URL tokenIntrospectionUrl, String clientId, String clientSecret, String token,
+            SSLContext sslContext, HostnameVerifier hostnameVerifier) throws RealmUnavailableException {
+        Assert.checkNotNullParam("clientId", clientId);
```

Comment:
Please remove the 1st 2 checks since they are already done in the constructor, and I'd also drop this `sslContext` parameter and set up the sslContext in this method as there is also a same named property.